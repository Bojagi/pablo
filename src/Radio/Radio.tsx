import React, { forwardRef } from 'react';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { getComponentStyle, transitionTransformer } from '../styleHelpers/getComponentStyle';

import {
  BaseCheckable,
  BaseCheckableProps,
  CheckableBoxProps,
  CheckableHandleProps,
} from '../shared/BaseCheckable';
import { getCustomStyles } from '../utils/useCustomStyles';
import { baseStyle } from '../shared/baseStyle';

const radioBoxSize = (props) => css`
  calc(${getComponentStyle('radio.handleSize.{size}')(props)} + 2 * (${getComponentStyle(
    'radio.innerPadding.{size}'
  )(props)} + ${getComponentStyle('radio.borderWidth')(props)}px))
`;

const RadioBox = styled.div<CheckableBoxProps>`
  ${baseStyle}
  position: relative;
  width: ${radioBoxSize};
  height: ${radioBoxSize};
  border-radius: 50%;
  padding: ${getComponentStyle('radio.innerPadding.{size}')};
  background-color: ${getComponentStyle('radio.backgroundColor')};
  border: ${getComponentStyle('radio.borderWidth')}px solid
    ${getComponentStyle('radio.borderColor')};
  opacity: ${(props) => (props.disabled ? 0.3 : 1)};
  transition: ${getComponentStyle('radio.boxTransition', transitionTransformer)};

  ${(props) =>
    props.focus &&
    css`
      box-shadow: 0 0 0 ${getComponentStyle('radio.focus.outlineSize')(props)}
        ${getComponentStyle('radio.focus.outlineColor')(props)};
    `};
  }
  ${getCustomStyles('radio.styles', 'box')}
`;

const RadioHandle = styled.div<CheckableHandleProps>`
  ${baseStyle}
  width: ${getComponentStyle('radio.handleSize.{size}')};
  height: ${getComponentStyle('radio.handleSize.{size}')};
  transform: scale(${(props: any) => (props.checked ? 1 : 0)});
  border-radius: 50%;
  transition: ${getComponentStyle('radio.handleTransition', transitionTransformer)};
  background-color: ${getComponentStyle('radio.handleColor')};
  ${getCustomStyles('radio.styles', 'handle')}
`;

export interface RadioProps extends BaseCheckableProps {
  value: string;
}

export const Radio = forwardRef<HTMLDivElement, RadioProps>((props: RadioProps, ref) => (
  <BaseCheckable
    ref={ref}
    role="radio"
    componentName="radio"
    componentType="radio"
    componentBox={RadioBox}
    componentHandle={RadioHandle}
    {...props}
  />
));
