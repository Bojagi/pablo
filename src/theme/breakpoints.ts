export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl';

export const breakpointNames: Breakpoint[] = ['sm', 'md', 'lg', 'xl'];

export type BreakpointsArray = [number, number, number, number];

export type Breakpoints = BreakpointsArray &
  Record<Breakpoint, number> & {
    breakpointNames: Breakpoint[];
  };

export function createBreakpoints(
  array: BreakpointsArray,
  usedBreakpointNames: Breakpoint[] = breakpointNames
): Breakpoints {
  const newBreakpoints = ([...array] as unknown) as Breakpoints;
  usedBreakpointNames.forEach((bp, index) => {
    newBreakpoints[bp] = newBreakpoints[index];
  });
  newBreakpoints.breakpointNames = usedBreakpointNames;

  return newBreakpoints as Breakpoints;
}

export const defaultBreakpoints: BreakpointsArray = [600, 1000, 1200, 1920];

export const breakpoints = createBreakpoints(defaultBreakpoints, breakpointNames);
