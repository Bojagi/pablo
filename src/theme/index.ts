export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
