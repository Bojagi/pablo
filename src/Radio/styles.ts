import { getColor } from '../styleHelpers/getColor';
import { getSpacing } from '../styleHelpers/getSpacing';
import { Style } from '../theme/types';
import { checkableBaseStyles, CheckableBaseStyles } from '../shared/checkableBaseStyles';

export interface RadioStyles extends CheckableBaseStyles {
  handleColor: Style;
  groupItemGap: {
    medium: Style;
    small: Style;
  };
}

export const radioStyles: RadioStyles = {
  ...checkableBaseStyles,
  handleColor: getColor('brand'),
  groupItemGap: {
    medium: getSpacing(5),
    small: getSpacing(4),
  },
};
