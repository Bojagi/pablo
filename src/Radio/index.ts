import './styles';

export * from './Radio';
export * from './RadioGroup';
