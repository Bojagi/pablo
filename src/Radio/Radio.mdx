---
name: Radio
route: /radio
menu: Components
---

import { Props, Playground } from 'docz'
import { Box } from '../Box/Box.tsx'
import { Radio } from './Radio.tsx'
import { RadioGroup } from './RadioGroup.tsx'
import { css } from 'styled-components';
import { getSpacing } from '../styleHelpers/index.ts'

# Radio

Form element that allows users to select a single item from a set of options. 
The `Radio` component shares an interface with the `Checkbox` and `Switch` components.

## Examples

### Single radio component

<Playground>
  {() => {
    const [checked, setChecked] = React.useState(false);
    return (
      <Radio checked={checked} onChange={() => setChecked(!checked)}  />
    );
  }}
</Playground>

### Radio selectors with labels

You can pass an optional `label` prop with a `ReactNode` value to show a label next to the checkbox.

<Playground>
  {() => {
    const [selectedDish, setDish] = React.useState(undefined);
    return (
      <Box display="flex" mx={-2}>
        <Radio mx={2} label="Spaghetti" checked={selectedDish === 'spaghetti'} onChange={() => setDish('spaghetti')} />
        <Radio mx={2} label="Pizza" checked={selectedDish === 'pizza'} onChange={() => setDish('pizza')} />
      </Box>
    );
  }}
</Playground>

### Grouped radio selectors

In the above example the selection management is done manually, it makes more sense though to use the `RadioGroup` to
group the values together. The `RadioGroup` component takes the selected `value` and `onChange` callback props.
For this to work the `Radio` components need to be passed a `value` where they should be selected. The `checked` prop is not needed.

<Playground>
  {() => {
    const [selectedDish, setDish] = React.useState(undefined);
    return (
      <RadioGroup value={selectedDish} onChange={setDish}>
        <Radio mx={2} label="Spaghetti" value="spaghetti" />
        <Radio mx={2} label="Pizza" value="pizza" />
      </RadioGroup>
    );
  }}
</Playground>

### Small radio selectors

You can pass the `size` prop with the `small` value which renders a smaller version.

<Playground>
  {() => {
    const [checked, setChecked] = React.useState(false);
    return (
      <Radio size="small" checked={checked} onChange={() => setChecked(!checked)} />
    );
  }}
</Playground>

When you want to set the size on a `RadioGroup` you have to set it in the `RadioGroup` component

<Playground>
  {() => {
    const [selectedDish, setDish] = React.useState(undefined);
    return (
      <RadioGroup size="small" value={selectedDish} onChange={setDish}>
        <Radio mx={2} label="Spaghetti" value="spaghetti" />
        <Radio mx={2} label="Pizza" value="pizza" />
      </RadioGroup>
    );
  }}
</Playground>

### Custom styling

You can overwrite the styles of parts of this component as well. 

<Playground>
  {() => {
    const [checked, setChecked] = React.useState(false);
    const customStyles = {
      box: css`
        background-color: blue;
        order: 2;
      `,
      handle: css`
        background-color: red;
      `,
      label: css`
        text-decoration: underline;
        order: 1;
        margin-right: ${getSpacing(4)};
      `,
    };
    return (
      <Radio
        checked={checked}
        label="weird"
        onChange={() => setChecked(!checked)}
        customStyles={customStyles} 
      />
    );
  }}
</Playground>

## Import

```js
import { Radio, RadioGroup } from '@bojagi/pablo/Radio';
```

## Props
<Props of={Radio} />
