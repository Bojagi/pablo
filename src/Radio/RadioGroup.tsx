import React from 'react';
import styled from 'styled-components';
import { Box, BoxProps } from '../Box';
import { guaranteeArray } from '../utils/guaranteeArray';
import { getComponentStyle } from '../utils/styleHelpers/getComponentStyle';
import { RadioProps, RadioSize } from './Radio';

const RadioGroupItem = styled.div<{ size: RadioSize }>`
  margin-bottom: ${getComponentStyle('radio.groupItemGap.{size}')};
`;

export interface RadioGroupProps extends BoxProps {
  value?: string;
  onChange: (value: string) => void;
  children: React.ReactElement<RadioProps> | React.ReactElement<RadioProps>[];
  size?: RadioSize;
  name?: string;
}

export function RadioGroup({
  value,
  onChange,
  name,
  children,
  size = 'medium',
  ...props
}: RadioGroupProps) {
  return (
    <Box role="radiogroup" {...props}>
      {guaranteeArray(children).map((child) => (
        <RadioGroupItem key={child.props.value} size={size}>
          {React.cloneElement(child, {
            checked: value === child.props.value,
            size,
            name,
            onChange: () => {
              onChange(child.props.value);
            },
          })}
        </RadioGroupItem>
      ))}
    </Box>
  );
}
