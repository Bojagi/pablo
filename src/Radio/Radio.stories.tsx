import React from 'react';
import { Box } from '../Box';
import { Radio, RadioProps } from './Radio';
import { RadioGroup } from './RadioGroup';

export default {
  title: 'Radio',
};

const ControlledRadio = ({ checked: checkedInitial, ...props }: RadioProps) => {
  const [checked, setChecked] = React.useState(checkedInitial);
  return (
    <Radio checked={checked} onChange={() => {}} onClick={() => setChecked(!checked)} {...props} />
  );
};

const baseStory = (args) => (
  <Box flex mx={-1} mb={1}>
    <ControlledRadio mx={1} value="1" name="hans" checked={true} {...args} />
    <ControlledRadio mx={1} value="2" name="hans" checked={false} {...args} />
    <ControlledRadio mx={1} value="3" name="hans" checked={true} disabled={true} {...args} />
  </Box>
);

export const Medium = baseStory.bind(null);
Medium.args = {
  size: 'medium',
};

export const Small = baseStory.bind(null);
Small.args = {
  size: 'small',
};

const BaseRadioGroupStory = (args) => {
  const [value, setValue] = React.useState<string>();
  return (
    <RadioGroup value={value} onChange={setValue} {...args}>
      <Radio value="hans" label="hans" />
      <Radio value="wurst" label="wurst" />
    </RadioGroup>
  );
};

export const RadioGroupMedium = BaseRadioGroupStory.bind(null);
RadioGroupMedium.args = {
  size: 'medium',
};

export const RadioGroupSmall = BaseRadioGroupStory.bind(null);
RadioGroupSmall.args = {
  size: 'small',
};
