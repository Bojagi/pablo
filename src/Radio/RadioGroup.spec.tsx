import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';
import { PabloThemeProvider } from '../theme';
import { RadioGroup } from './RadioGroup';
import { Radio } from './Radio';
import { clampRegExp } from '../../testUtils/matchClamp';

let onChangeMock;

beforeEach(() => {
  onChangeMock = vi.fn();
});

test('Work with one radio button', () => {
  const { container } = render(
    <PabloThemeProvider>
      <RadioGroup onChange={onChangeMock} name="greeting">
        <Radio id="custom-id" label="Hello" value="hello" />
      </RadioGroup>
    </PabloThemeProvider>
  );
  expect(container).toMatchSnapshot();
});

test('Render with nothing selected', () => {
  const { container, getAllByTestId } = renderComponent({ name: 'greeting' });
  expect(container).toMatchSnapshot();
  const allRadios = getAllByTestId('pbl-radio-input');
  expect(allRadios).toHaveLength(3);
  expect(allRadios[0]).not.toBeChecked();
  expect(allRadios[1]).not.toBeChecked();
  expect(allRadios[2]).not.toBeChecked();
  expect(allRadios[0]).toHaveAttribute('name', 'greeting');
  expect(allRadios[1]).toHaveAttribute('name', 'greeting');
  expect(allRadios[2]).toHaveAttribute('name', 'greeting');
});

test('Render with "world" selected', () => {
  const { getAllByTestId } = renderComponent({ value: 'world' });
  const allRadios = getAllByTestId('pbl-radio-input');
  expect(allRadios).toHaveLength(3);
  expect(allRadios[0]).not.toBeChecked();
  expect(allRadios[1]).toBeChecked();
  expect(allRadios[2]).not.toBeChecked();
});

test('Click on "bye" and trigger "bye" being selected', () => {
  const { getByText, getAllByTestId, rerender } = renderComponent({ value: 'world' });
  const byeLabel = getByText('Bye');

  // Click on label to trigger 'bye' being checked
  expect(onChangeMock).toHaveBeenCalledTimes(0);
  act(() => {
    fireEvent.click(byeLabel as any);
  });
  expect(onChangeMock).toHaveBeenCalledTimes(1);
  expect(onChangeMock).toHaveBeenCalledWith('bye');

  // Rerender with value being 'bye'
  rerender(getRenderHtml({ value: 'bye' }));

  // Now radios should have been updated
  const allRadios = getAllByTestId('pbl-radio-input');
  expect(allRadios).toHaveLength(3);
  expect(allRadios[0]).not.toBeChecked();
  expect(allRadios[1]).not.toBeChecked();
  expect(allRadios[2]).toBeChecked();
});

test('Render with size "medium"', () => {
  const { getAllByTestId } = renderComponent({ size: 'medium' });
  const allRadios = getAllByTestId('pbl-radio-handle');
  expect(allRadios).toHaveLength(3);
  expect(allRadios[0]).toHaveStyleRule('width', clampRegExp(1.5, 0.5, 1));
  expect(allRadios[1]).toHaveStyleRule('width', clampRegExp(1.5, 0.5, 1));
  expect(allRadios[2]).toHaveStyleRule('width', clampRegExp(1.5, 0.5, 1));
});

test('Render with size "small"', () => {
  const { getAllByTestId } = renderComponent({ size: 'small' });
  const allRadios = getAllByTestId('pbl-radio-handle');
  expect(allRadios).toHaveLength(3);
  expect(allRadios[0]).toHaveStyleRule('width', clampRegExp(1, 0.5, 1));
  expect(allRadios[1]).toHaveStyleRule('width', clampRegExp(1, 0.5, 1));
  expect(allRadios[2]).toHaveStyleRule('width', clampRegExp(1, 0.5, 1));
});

function renderComponent(props) {
  return render(getRenderHtml(props));
}

function getRenderHtml(props) {
  return (
    <PabloThemeProvider>
      <RadioGroup onChange={onChangeMock} {...props}>
        <Radio id="custom-id-1" label="Hello" value="hello" />
        <Radio id="custom-id-2" label="World" value="world" />
        <Radio id="custom-id-3" label="Bye" value="bye" />
      </RadioGroup>
    </PabloThemeProvider>
  );
}
