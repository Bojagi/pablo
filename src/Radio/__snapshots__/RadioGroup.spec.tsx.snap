// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render with nothing selected 1`] = `
.emotion-0 {
  box-sizing: border-box;
}

.emotion-1 {
  box-sizing: border-box;
  margin-bottom: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  box-sizing: border-box;
  position: relative;
  width: calc(clamp(0.75rem, 0.5357142857142857rem + 1.0714285714285714vw, 1.5rem) + 2 * (clamp(0.25rem, 0.17857142857142858rem + 0.35714285714285715vw, 0.5rem) + 1px));
  height: calc(clamp(0.75rem, 0.5357142857142857rem + 1.0714285714285714vw, 1.5rem) + 2 * (clamp(0.25rem, 0.17857142857142858rem + 0.35714285714285715vw, 0.5rem) + 1px));
  border-radius: 50%;
  padding: clamp(0.25rem, 0.17857142857142858rem + 0.35714285714285715vw, 0.5rem);
  background-color: var(--pbl-theme-colors-common-white);
  border: 1px solid var(--pbl-theme-colors-borders-main);
  opacity: 1;
  -webkit-transition: box-shadow 0.3s ease-in-out;
  transition: box-shadow 0.3s ease-in-out;
}

.emotion-4 {
  box-sizing: border-box;
  width: clamp(0.75rem, 0.5357142857142857rem + 1.0714285714285714vw, 1.5rem);
  height: clamp(0.75rem, 0.5357142857142857rem + 1.0714285714285714vw, 1.5rem);
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  border-radius: 50%;
  -webkit-transition: -webkit-transform 0.15s cubic-bezier(0.34, 1.56, 0.64, 1);
  transition: transform 0.15s cubic-bezier(0.34, 1.56, 0.64, 1);
  background-color: var(--pbl-theme-colors-brand-main);
}

.emotion-5 {
  box-sizing: border-box;
  position: absolute;
  display: block;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
}

.emotion-6 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
  margin-left: clamp(0.5rem, 0.4285714285714286rem + 0.35714285714285715vw, 0.75rem);
}

.emotion-6 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-6 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <div
    class="emotion-0"
    role="radiogroup"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
        role="radio"
      >
        <div
          class="emotion-3"
          data-testid="pbl-radio"
        >
          <div
            class="emotion-4"
            data-testid="pbl-radio-handle"
          />
          <input
            class="emotion-5"
            data-testid="pbl-radio-input"
            id="custom-id-1"
            name="greeting"
            type="radio"
            value="hello"
          />
        </div>
        <label
          class="emotion-6"
          data-testid="pbl-radio-label"
          for="custom-id-1"
        >
          Hello
        </label>
      </div>
    </div>
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
        role="radio"
      >
        <div
          class="emotion-3"
          data-testid="pbl-radio"
        >
          <div
            class="emotion-4"
            data-testid="pbl-radio-handle"
          />
          <input
            class="emotion-5"
            data-testid="pbl-radio-input"
            id="custom-id-2"
            name="greeting"
            type="radio"
            value="world"
          />
        </div>
        <label
          class="emotion-6"
          data-testid="pbl-radio-label"
          for="custom-id-2"
        >
          World
        </label>
      </div>
    </div>
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
        role="radio"
      >
        <div
          class="emotion-3"
          data-testid="pbl-radio"
        >
          <div
            class="emotion-4"
            data-testid="pbl-radio-handle"
          />
          <input
            class="emotion-5"
            data-testid="pbl-radio-input"
            id="custom-id-3"
            name="greeting"
            type="radio"
            value="bye"
          />
        </div>
        <label
          class="emotion-6"
          data-testid="pbl-radio-label"
          for="custom-id-3"
        >
          Bye
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Work with one radio button 1`] = `
.emotion-0 {
  box-sizing: border-box;
}

.emotion-1 {
  box-sizing: border-box;
  margin-bottom: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  box-sizing: border-box;
  position: relative;
  width: calc(clamp(0.75rem, 0.5357142857142857rem + 1.0714285714285714vw, 1.5rem) + 2 * (clamp(0.25rem, 0.17857142857142858rem + 0.35714285714285715vw, 0.5rem) + 1px));
  height: calc(clamp(0.75rem, 0.5357142857142857rem + 1.0714285714285714vw, 1.5rem) + 2 * (clamp(0.25rem, 0.17857142857142858rem + 0.35714285714285715vw, 0.5rem) + 1px));
  border-radius: 50%;
  padding: clamp(0.25rem, 0.17857142857142858rem + 0.35714285714285715vw, 0.5rem);
  background-color: var(--pbl-theme-colors-common-white);
  border: 1px solid var(--pbl-theme-colors-borders-main);
  opacity: 1;
  -webkit-transition: box-shadow 0.3s ease-in-out;
  transition: box-shadow 0.3s ease-in-out;
}

.emotion-4 {
  box-sizing: border-box;
  width: clamp(0.75rem, 0.5357142857142857rem + 1.0714285714285714vw, 1.5rem);
  height: clamp(0.75rem, 0.5357142857142857rem + 1.0714285714285714vw, 1.5rem);
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  border-radius: 50%;
  -webkit-transition: -webkit-transform 0.15s cubic-bezier(0.34, 1.56, 0.64, 1);
  transition: transform 0.15s cubic-bezier(0.34, 1.56, 0.64, 1);
  background-color: var(--pbl-theme-colors-brand-main);
}

.emotion-5 {
  box-sizing: border-box;
  position: absolute;
  display: block;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
}

.emotion-6 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-size: var(--pbl-theme-typography-body-fontSize);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
  margin-left: clamp(0.5rem, 0.4285714285714286rem + 0.35714285714285715vw, 0.75rem);
}

.emotion-6 variants bold {
  font-weight: var(--pbl-theme-typography-body-variants-bold-fontWeight);
}

.emotion-6 variants small {
  font-size: var(--pbl-theme-typography-body-variants-small-fontSize);
}

<div>
  <div
    class="emotion-0"
    role="radiogroup"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
        role="radio"
      >
        <div
          class="emotion-3"
          data-testid="pbl-radio"
        >
          <div
            class="emotion-4"
            data-testid="pbl-radio-handle"
          />
          <input
            class="emotion-5"
            data-testid="pbl-radio-input"
            id="custom-id"
            name="greeting"
            type="radio"
            value="hello"
          />
        </div>
        <label
          class="emotion-6"
          data-testid="pbl-radio-label"
          for="custom-id"
        >
          Hello
        </label>
      </div>
    </div>
  </div>
</div>
`;
